eng/common/scripts/Submit-PullRequest.ps1

```diff
@@ -49,40 +49,90 @@ param(
   [string]$PRBody = $PRTitle,
 
   [Parameter(Mandatory = $false)]
-  [string]$PRLabels
+  [string]$PRLabels,
+
+  [Parameter(Mandatory = $false)]
+  [string]$UserReviewers,
+
+  [Parameter(Mandatory = $false)]
+  [string]$TeamReviewers,
+
+  [Parameter(Mandatory = $false)]
+  [string]$Assignees,
+
+  $PRNumber
 )
 
+$baseURI = "https://api.github.com/repos"
+function SplitMembers ($membersString)
+{
+  return @($membersString.Split(",") | % { $_.Trim() } | ? { return $_ })
+}
+
+$userAdditions = SplitMembers -membersString $UserReviewers
+$teamAdditions = SplitMembers -membersString $TeamReviewers
+$labelAdditions = SplitMembers -membersString $PRLabels
+$assigneeAdditions = SplitMembers -membersString $Assignees
+
 $headers = @{
   Authorization = "bearer $AuthToken"
 }
 
-$query = "state=open&head=${PROwner}:${PRBranch}&base=${BaseBranch}"
-
-function AddLabels([int] $prNumber, [string] $prLabelString)
-{
-  # Adding labels to the pr.
-  if (-not $prLabelString) {
-    Write-Verbose "There are no labels added to the PR."
-    return
+function AddMembers($apiURI, $memberName, $additionSet, $body) {
+  $headers = @{
+    Authorization = "bearer $AuthToken"
   }
+  $errorOccurred = $false
 
-  # Parse the labels from string to array
-  $prLabelArray = @($prLabelString.Split(",") | % { $_.Trim() } | ? { return $_ })
-  $prLabelUri = "https://api.github.com/repos/$RepoOwner/$RepoName/issues/$prNumber"
-  $labelRequestData = @{
-    labels = $prLabelArray
-  }
   try {
-    $resp = Invoke-RestMethod -Method PATCH -Headers $headers $prLabelUri -Body ($labelRequestData | ConvertTo-Json)
+    if (-not $body)
+    {
+      $body = @{}
+      $body[$memberName] = @($additionSet)
+    }
+
+    $resp = Invoke-RestMethod -Method 'Post' -Headers $headers -Body ($body | ConvertTo-Json) -Uri $apiURI -MaximumRetryCount 3
+    $resp | Write-Verbose
   }
   catch {
-    Write-Error "Invoke-RestMethod $prLabelUri failed with exception:`n$_"
+    Write-Error "Invoke-RestMethod $apiURI failed with exception:`n$_"
+    $errorOccurred = $true
   }
 
-  $resp | Write-Verbose
-  Write-Host -f green "Label(s) [$prLabelArray] added to pull request: https://github.com/$RepoOwner/$RepoName/pull/$prNumber"
+  return $errorOccurred
+}
+
+function AddReviewers ($prNumber) {
+  $uri = "$baseURI/$RepoOwner/$RepoName/pulls/$prNumber/requested_reviewers"
+  if ($userAdditions) {
+    $errorsOccurredAddingUsers = AddMembers -apiURI $uri -memberName "reviewers" -additionSet $userAdditions
+    if ($errorsOccurredAddingUsers) { exit 1 }
+    Write-Host -f green "User(s) [$userAdditions] added to: https://github.com/$RepoOwner/$RepoName/issue/$prNumber"
+  }
+  if ($teamAdditions) {
+    $errorsOccurredAddingTeams = AddMembers -apiURI $uri -memberName "team_reviewers" -additionSet $teamAdditions
+    if ($errorsOccurredAddingTeams) { exit 1 }
```

Comment:
We should generally avoid exiting a script in the middle of a function.  We should either throw an exception and handle those at the calling site or return an error and handle that at the calling site.